### PR TITLE
super/cc: Fix --cc=llvm_clang invoked as clang

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fix the error:
```
brew sh --cc=llvm_clang <<<'clang --version'
Failed to execute llvm_clang --version
```

### CODE REVIEW

File: Library/Homebrew/shims/super/cc [hunk 1]
```diff
@@ -72,7 +72,11 @@ class Cmd
     else
       # Note that this is a universal fallback, so that we'll always invoke
       # HOMEBREW_CC regardless of what name under which the tool was invoked.
-      ENV["HOMEBREW_CC"]
+      if ENV["HOMEBREW_CC"] == "llvm_clang"
+        "#{ENV["HOMEBREW_PREFIX"]}/opt/llvm/bin/clang"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/shims/super/cc]: This looks like it should be handled by line 61; any idea why it isn't?

Comment (sjackman) [inline Library/Homebrew/shims/super/cc]: Yep. This patch addresses the case when `HOMEBREW_CC == llvm_clang` and `super/cc` is invoked as any compiler other than `llvm_clang`, for example `clang` or `gcc`. Line 61 tests whether `arg0 == llvm_clang` but not `HOMEBREW_CC == llvm_clang`.

Comment (MikeMcQuaid) [inline Library/Homebrew/shims/super/cc]: Gotcha, thanks.

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman!

**Comment by aw1621107** [comment]
This might fix #2237

**Pull request merged**